simulation/halsim_ws_core/src/main/native/cpp/WSHalProviders.cpp

```diff
@@ -0,0 +1,57 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2017-2020 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#include "WSHalProviders.h"
+
+namespace wpilibws {
+
+void HALSimWSHalProvider::OnNetworkConnected(
+    std::shared_ptr<HALSimBaseWebSocketConnection> ws) {
+  {
+    std::lock_guard lock(mutex);
+
+    // store a weak reference to the websocket object
+    m_ws = ws;
```

Comment:
The client should receive all values when they first connect, since there isn't a way for it to ask for values. When you removed the diff mechanism, this was broken.

Reply:
Got it. I have a plan for adding this capability back in. Will add a new commit tomorrow